code/game/gamemodes/objectives_subtypes.dm

```diff
@@ -0,0 +1,20 @@
+/datum/objective/assassinate/mindshielded
+	name = "Assassinate mindshielded"
+	flags_target = MINDSHIELDED_TARGET
+
+/datum/objective/assassinate/nomindshield
+	name = "Assassinate non-mindshielded"
+	flags_target = UNMINDSHIELDED_TARGET
+
+/datum/objective/assassinate/syndicate
+	name = "Assassinate syndicate agent"
+	flags_target = SYNDICATE_TARGET
+
+/datum/objective/assassinate/syndicate/update_explanation_text()
+	..()
+	if(target?.current)
+		explanation_text = "Assassinate [target.current.real_name], the Syndicate agent undercover as the [target.assigned_role]."
```

Comment:
I am personally not a fan of this (Especially since it's not documented in the PR description, please do add it.) I don't think traitors should automatically be aware if their target is a traitor; Currently, as an agent, you can choose not to use your codewords, and stay hidden, meaning that you are treated differently as a target than when you are a known traitor.

Reply:
This only applies to hunter syndicates, who are 10 percent of syndicate agents, not all syndies. PR description has been updated to be more explicit on this, this was documented but unclear.